### PR TITLE
Fix loosing hidden columns

### PR DESCRIPTION
**Pull request by JosephMarinier**
## Description:

Fix loosing hidden columns when sorting Performance Analysis table. Controlling the sort model triggers re-renders that lose any uncontrolled states, so we must control all states, including `columnVisibilityModel`.

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **FRONTEND TYPES.** Regenerate the front-ent types if you played with types and routes.
  Run `cd webapp && yarn types` while the back-end is running.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

File: webapp/src/components/Metrics/PerformanceAnalysis.tsx [hunk 1]
```diff
@@ -273,6 +278,8 @@ const PerformanceAnalysis: React.FC<Props> = ({ jobId, pipeline }) => {
         }}
         sortModel={sortModel}
         onSortModelChange={setSortModel}
+        columnVisibilityModel={columnVisibilityModel}
```
[the remark this answers is not in the document]
Comment (JosephMarinier) [reply webapp/src/components/Metrics/PerformanceAnalysis.tsx:281]: Does my comment on lines 89-90 help? Any specific question? Idea to make it clearer?
https://github.com/ServiceNow/azimuth/blob/7f58774e8bd13b795ff36376b5edb65ffcd8f459/webapp/src/components/Metrics/PerformanceAnalysis.tsx#L89-L90
And of "all states" we are currently using the `columnVisibilityModel` state.

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,5 +20,6 @@ Released changes are shown in the
 ### Removed
 
 ### Fixed
+- Fixed loosing hidden columns when sorting Performance Analysis table.
```
Comment (christyler3030) [inline CHANGELOG.md:23]: Nitpick: loosing -> losing

File: webapp/src/components/Metrics/PerformanceAnalysis.tsx [hunk 3]
```diff
@@ -87,10 +88,14 @@ const PerformanceAnalysis: React.FC<Props> = ({ jobId, pipeline }) => {
     ...pipeline,
   });
 
-  // Track table sort model to keep 'overall' at top
+  // Track table sort model to keep 'overall' at top.
```
Comment (christyler3030) [inline webapp/src/components/Metrics/PerformanceAnalysis.tsx:91]: I know this is not part of this PR but does this actually keep 'overall' at the top? how does it interact with the `customSort` below?

File: webapp/src/components/Metrics/PerformanceAnalysis.tsx [hunk 4]
```diff
@@ -87,10 +88,14 @@ const PerformanceAnalysis: React.FC<Props> = ({ jobId, pipeline }) => {
...
+  // Controlling sort model triggers rerenders that lose any uncontrolled states,
+  // so we must control all states.
   const [sortModel, setSortModel] = React.useState<GridSortModel>([
     { field: "utteranceCount", sort: "desc" },
   ]);
+  const [columnVisibilityModel, onColumnVisibilityModelChange] =
```
Comment (christyler3030) [inline webapp/src/components/Metrics/PerformanceAnalysis.tsx:97]: Let's respect the naming convention even if it is not strictly enforced.

File: webapp/src/components/Metrics/PerformanceAnalysis.tsx (hunk 3 above)
Comment (JosephMarinier) [reply webapp/src/components/Metrics/PerformanceAnalysis.tsx:91]: Yes, our `customSort` uses `sortModel` to determine at which "end" of the list it should return the `overall` row base on whether the sort model is in ascending or descending order, to always keep the `overall` row on top.

https://github.com/ServiceNow/azimuth/blob/57990d544bec30cfef58ac4c9085aede837c0c42/webapp/src/components/Metrics/PerformanceAnalysis.tsx#L128-L131

File: CHANGELOG.md (hunk 2 above)
Comment (JosephMarinier) [reply CHANGELOG.md:23]: Oupsyyy! Thank you!

**Comment by JosephMarinier** [comment]
> And quite easily it turns out.

We it turns out getting a clean solution was not thaaat easy! Thank you @christyler3030. Final review?

File: webapp/src/components/Metrics/PerformanceAnalysis.tsx [hunk 5]
```diff
@@ -81,16 +85,22 @@ const PerformanceAnalysis: React.FC<Props> = ({ jobId, pipeline }) => {
...
     jobId,
     datasetSplitName: selectedDatasetSplit,
     ...pipeline,
   });
 
-  // Track table sort model to keep 'overall' at top
-  const [sortModel, setSortModel] = React.useState<GridSortModel>([
-    { field: "utteranceCount", sort: "desc" },
-  ]);
+  // Track table sort model to keep 'overall' at top.
```
Comment (christyler3030) [inline webapp/src/components/Metrics/PerformanceAnalysis.tsx:96]: Nitpick: no longer tracking whole sort model, only sort direction.

**Review by christyler3030 — APPROVED** [review]

File: webapp/src/components/Metrics/PerformanceAnalysis.tsx (hunk 5 above)
Comment (JosephMarinier) [reply webapp/src/components/Metrics/PerformanceAnalysis.tsx:96]: True that! Even if that's your original comment. 😜 I'll rephrase. Thanks!

**Pull request merged**